### DISCUSSION TITLE
Had to Change mqtt_room in Home Assistant Back to the Old Format

### DISCUSSION THREAD

**Discussion by gcortes** [ESPresense/ESPresense#1204] [Q&A]
I have my iPhone configured in the Home Assistant config. I used to have it set to device_id: 'apple:1006:10-12'. I then changed it to use "irk:...". I believe I did this from reading about it in an upgrade at some point. It worked fine for a year or more.
I then changed routers and Mosquitto brokers at about the same time and the phone location function stopped working. Since it was appearing is MQTT Explorer as espresense/devices/apple:1006:10-12, I tried changing it back to that id and it started working again. Should it have always been set to device_id: 'apple:1006:10-12' or was IRK used for a while and then changed back?

**Comment by Terastar-Paperclip** [comment]
Sorry this sat quiet. The device_id format change landing at the same time as a broker/router swap makes this hard to diagnose from the description alone — flagging it for the core team to take a proper look rather than guessing. Will follow up here.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here. Traced through the current identification code (`BleFingerprint::fingerprintAddress`) to answer this properly.

Both formats are real and both are still supported today, but they're not the same identity path:

- **`apple:1006:10-12`** is a signature derived from Apple's Nearby-Action continuity manufacturer data (message type + byte length). It's coarse — it can be shared by multiple iPhones of the same model/OS generation — but it needs nothing from the node besides seeing the advertisement.
- **`irk:<hex>`** is your phone's actual per-device Identity Resolving Key, learned via BLE enrollment (pairing) and stored on the node. It's the precise, unique identifier, but it only resolves if the node still has that IRK stored. If a node loses its enrolled IRK list — most commonly from a factory reset or reflash — it silently falls back down the chain to the coarser `apple:` signature instead of erroring.

Your router/broker swap is almost certainly a coincidence, not the cause. The more likely trigger is that a node got reflashed or reset around the same time and lost its enrolled IRK for your phone. Practical fix: re-run enrollment (the Enroll button in the node's web UI) to relearn the IRK, then switch your HA `device_id` back to `irk:...`. If you'd rather not re-enroll, staying on the `apple:` id is fine — just know it's less selective (could theoretically match another same-model iPhone in range).
